integration/flink/app/build.gradle

```diff
@@ -0,0 +1,368 @@
+    compileOnly "org.apache.flink:flink-table-common:$flinkVersion"
+    compileOnly "org.apache.flink:flink-connector-kafka:${versions.kafka}"
+    compileOnly "org.apache.flink:flink-connector-jdbc:${versions.jdbc}"
+    compileOnly "org.apache.flink:flink-avro:$flinkVersion"
+    compileOnly "org.apache.flink:flink-avro-confluent-registry:$flinkVersion"
+    compileOnly "org.apache.iceberg:iceberg-flink-runtime-${versions.alternativeShort}:$icebergVersion"
+    compileOnly "org.apache.flink:flink-connector-cassandra_2.12:${versions.cassandra}"
+
+    annotationProcessor "org.projectlombok:lombok:${lombokVersion}"
+
+    implementation 'org.javassist:javassist:3.30.2-GA'
+    implementation 'org.apache.httpcomponents.client5:httpclient5:5.3.1'
+
+    testImplementation "org.apache.flink:flink-java:$flinkVersion"
+    testImplementation "org.apache.flink:flink-streaming-java:$flinkVersion"
+    testImplementation "org.apache.flink:flink-runtime-web:$flinkVersion"
+    testImplementation "org.apache.flink:flink-connector-base:$flinkVersion"
+    testImplementation "org.apache.flink:flink-connector-kafka:${versions.kafka}"
+    testImplementation "org.apache.flink:flink-connector-jdbc:${versions.jdbc}"
+    testImplementation "org.apache.flink:flink-connector-cassandra_2.12:${versions.cassandra}"
+    testImplementation "org.apache.flink:flink-table-common:$flinkVersion"
+    testImplementation "org.apache.flink:flink-avro:$flinkVersion"
+    testImplementation "org.apache.flink:flink-avro-confluent-registry:$flinkVersion"
+    testImplementation "org.apache.iceberg:iceberg-flink-runtime-${versions.alternativeShort}:$icebergVersion"
+    testImplementation platform('org.junit:junit-bom:5.10.2')
+    testImplementation 'org.hamcrest:hamcrest-library:2.2'
+    testImplementation "org.testcontainers:mockserver:${testcontainersVersion}"
+    testImplementation "org.testcontainers:junit-jupiter:${testcontainersVersion}"
+    testImplementation 'org.mock-server:mockserver-client-java:5.12.0'
+    testImplementation group: 'org.awaitility', name: 'awaitility', version: '4.2.0'
+    testImplementation 'com.github.tomakehurst:wiremock:3.0.1'
+    testImplementation 'org.apache.logging.log4j:log4j-slf4j-impl:2.22.1'
+
+    testImplementation "org.assertj:assertj-core:${assertjVersion}"
+    testImplementation "org.junit.jupiter:junit-jupiter:${junit5Version}"
+    testImplementation "org.junit.jupiter:junit-jupiter-params:${junit5Version}"
+    testImplementation "org.mockito:mockito-core:${mockitoVersion}"
+    testImplementation "org.mockito:mockito-inline:${mockitoVersion}"
+    testImplementation "org.mockito:mockito-junit-jupiter:${mockitoVersion}"
+    testImplementation "org.projectlombok:lombok:${lombokVersion}"
+
+    testAnnotationProcessor "org.projectlombok:lombok:${lombokVersion}"
+}
+
+sourceSets {
+    main.java.srcDirs = ["src/main/java"]
+    test.java.srcDirs = ["src/test/java"]
+}
```

Comment:
This is unnecessary. The `java-library` plugin will already creates the `main` and `test` source sets, and it configures them. This can be safely removed.